browser/ui/browser_commands.cc

```diff
@@ -863,6 +863,35 @@ void CloseDuplicateTabs(Browser* browser) {
   }
 }
 
+bool HasAnyDuplicateTabs(Browser* browser) {
+  auto* tsm = browser->tab_strip_model();
+  base::flat_set<GURL> urls;
+  for (int i = 0; i < tsm->GetTabCount(); ++i) {
+    auto* tab = tsm->GetWebContentsAt(i);
+    if (urls.contains(tab->GetVisibleURL())) {
+      return true;
+    }
+    urls.insert(tab->GetVisibleURL());
+  }
+
+  return false;
+}
+
+void CloseAllDuplicateTabs(Browser* browser) {
+  auto* tsm = browser->tab_strip_model();
+  base::flat_set<GURL> urls;
+  for (int i = 0; i < tsm->GetTabCount(); ++i) {
+    auto* tab = tsm->GetWebContentsAt(i);
+    if (urls.contains(tab->GetVisibleURL())) {
+      tab->Close();
+      --i;
+      continue;
+    }
+
+    urls.insert(tab->GetVisibleURL());
+  }
```

Comment:
How about reusing `ExecuteCloseTabsByIndicesCommand()`?
and it would be great if we have test cases for our new apis.